tests/facet_search.test.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  clearAllIndexes,
+  config,
+  getClient,
+} from './utils/meilisearch-test-utils'
+
+const index = {
+  uid: 'movies_test',
+}
+
+const dataset = [
+  {
+    id: 123,
+    title: 'Pride and Prejudice',
+    genres: ['romance', 'action'],
+  },
+  {
+    id: 456,
+    title: 'Le Petit Prince',
+    genres: ['adventure', 'comedy'],
+  },
+  {
+    id: 2,
+    title: 'Le Rouge et le Noir',
+    genres: 'romance',
+  },
+  {
+    id: 1,
+    title: 'Alice In Wonderland',
+    genres: ['adventure'],
+  },
+]
+
+describe.each([
+  { permission: 'Master' },
+  { permission: 'Admin' },
+  { permission: 'Search' },
+])('Test on POST search', ({ permission }) => {
+  beforeAll(async () => {
+    await clearAllIndexes(config)
+    const client = await getClient('Master')
+    const newFilterableAttributes = ['genres', 'title']
+    await client.createIndex(index.uid)
+    await client.index(index.uid).updateSettings({
+      filterableAttributes: newFilterableAttributes,
+    })
+    const { taskUid } = await client.index(index.uid).addDocuments(dataset)
+    await client.waitForTask(taskUid)
+  })
+
+  test(`${permission} key: basic facet value search`, async () => {
+    const client = await getClient(permission)
+
+    const params = {
+      facetQuery: 'a',
+      facetName: 'genres',
+    }
+    const response = await client.index(index.uid).searchForFacetValues(params)
+
+    expect(response.facetHits.length).toEqual(2)
+    expect(response.facetQuery).toEqual('a')
+  })
+
+  test(`${permission} key: facet value search with no facet query`, async () => {
+    const client = await getClient(permission)
+
+    const params = {
+      facetName: 'genres',
+    }
+    const response = await client.index(index.uid).searchForFacetValues(params)
+
+    expect(response.facetHits.length).toEqual(4)
+    expect(response.facetQuery).toEqual(null)
+  })
+
+  test(`${permission} key: facet value search with filter`, async () => {
+    const client = await getClient(permission)
+
+    const params = {
+      facetName: 'genres',
+      facetQuery: 'a',
```

Comment:
I'm thinking in the ergonomics of the method if, for some reason, the people using this method will try to use the same params in the `search` request. Unfortunately, this is just guesswork, but it would be cool if we just ignore the facet* arguments in the search query on the engine side.

Reply:
Sorry if that was already suggested/ignored before.

Reply:
Perfect! ❤️